types/denom.go

```diff
@@ -8,6 +8,9 @@ import (
 // multipliers (e.g. 1atom = 10^-6uatom).
 var denomUnits = map[string]Dec{}
```

Comment:
I think it would be ideal if this `denom` registry were managed within `x/bank` rather than a `types/` global. Somewhat unrelated to this PR, but noticing. Then it could be related to denom metadata stored in `x/bank`.

Reply:
I think it's more like a client side utility, rather than on chain logic, there should be only the base unit inside the chain.
where is the best places for this kind of logic?

Reply:
Right, so recently I wrote up an ADR where we provide denomination "metadata" on-chain. This metadata is purely informational.

While I do agree with @aaronc, it would not be ideal to have to query the state every time we want to convert coins. So I think this is fine for now, but something to keep in mind -- `denomUnits` must be equivalent to what is found on chain.